code/script.js

```diff
@@ -1,47 +1,235 @@
+         </div>
+       </section>
+     `
+   }
 
-// This function will add a chat bubble in the correct place based on who the sender is
-const showMessage = (message, sender) => {
-  if (sender === 'user') {
-    chat.innerHTML += `
-      <section class="user-msg">
-        <div class="bubble user-bubble">
-          <p>${message}</p>
-        </div>
-        <img src="assets/user.png" alt="User" />  
-      </section>
-    `
-  } else if (sender === 'bot') {
-    chat.innerHTML += `
-      <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
-        <div class="bubble bot-bubble">
-          <p>${message}</p>
-        </div>
-      </section>
-    `
-  }
-  // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
   chat.scrollTop = chat.scrollHeight
-}
+ }
 
 // Starts here
-const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
-}
-
-// Set up your eventlisteners here
-
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
-// This means the greeting function will be called one second after the website is loaded.
-setTimeout(greeting, 1000)
+ const greeting = () => {
+   showMessage(`Hello there, What's your name?`, 'bot')
+ }
+  
+// Just to check it out, change 'bot' to 'user' here 👆
+
+
+ const handleInput = (event) => {
+   currentQuestion++
+   event.preventDefault()
+
+   console.log('our currentQuestion variable is:', currentQuestion)
+   if (currentQuestion === 1) {
+    handleNameQuestion()}
+    else if (currentQuestion === 2) {
+      languageQuestion()}
+   }
+
+  // Questions asked 
+ 
+  const languageQuestion = (name) => {
+    showMessage(`Hi ${name}, Welcome by learning Swedish which subject do you want to learn?`, 'bot')
+
+    inputWrapper.innerHTML = `
+      <button id="food">Food🥗</button>
+      <button id="greetings">Greetings👋</button>
+      <button id="jobs">Jobs👨‍⚖️</button>
+    `
+    document 
+      .getElementById('food')
+      .addEventListener('click', () => {askForType('food')})
+    document
+      .getElementById('greetings')
+      .addEventListener('click', () => {askForType('greetings')})
+    document
+      .getElementById('jobs')
+      .addEventListener('click', () => {askForType('jobs')})
+
+      
+
+      currentQuestion++
+      console.log('Current question set to', currentQuestion)
+   }
+
+   
+
+  //Question Functions
+
+    const handleNameQuestion = () => {
+      const name = inputValue.value
+
+      showMessage(`My name is ${name}`, 'user')
+      inputValue.value = ''   
+
+      setTimeout(() => languageQuestion(name), 1500)
+    }
+
+   
+    const askForType = (subject) => {
+      if (subject === 'food') {
```

Comment:
Impressed that most of the bot<>user conversation happens within a single function. Really cool to see and seems to help keep everything in order, making it a bit easier to understand what will happen.
However, a few descriptions/comments in the code would help explain the function and its stages/flow, as it initially is a bit to grasp.